### PR TITLE
[Aks] Align the logic of Set-AzAks to cli

### PR DESCRIPTION
**Pull request by wyunchi-ms**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Aks/Aks/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Fixed the issue that `Set-AzAksCluster` can't upgrade cluster in some cases. [#14583]
```
Comment (dingmeng-xue) [inline src/Aks/Aks/ChangeLog.md:21]: do we know what case is?

Comment (wyunchi-ms) [reply src/Aks/Aks/ChangeLog.md:21]: When the version of node pool doesn't match cluster's version.

File: src/Aks/Aks/Properties/Resources.resx [hunk 2]
```diff
@@ -426,4 +426,22 @@
...
     <value>Do you want to reset the credential of the aks cluster?</value>
   </data>
+  <data name="ConfirmControlPlaneOnly" xml:space="preserve">
+    <value>Since control-plane-only argument is specified, this will upgrade only the control plane to {0}. Node pool will not change. Continue?</value>
+  </data>
+  <data name="ConfirmControlPlaneOnlyInVMASCluster" xml:space="preserve">
+    <value>Legacy clusters do not support control plane only upgrade. All node pools will be upgraded to {0} as well. Continue?</value>
+  </data>
+  <data name="ConfirmNotControlPlaneOnly" xml:space="preserve">
+    <value>Since control-plane-only argument is not specified, this will upgrade the control plane AND all nodepools to version {0}. Continue?</value>
```
Comment (isra-fel) [inline src/Aks/Aks/Properties/Resources.resx:436]: nodepools -> node pools

File: src/Aks/Aks/Commands/SetAzureRmAks.cs [hunk 3]
```diff
@@ -259,11 +266,81 @@ public override void ExecuteCmdlet()
                             }
                         }
 
+                        if (this.IsParameterBound(c => c.KubernetesVersion) && this.IsParameterBound(c => c.NodeImageOnly))
+                        {
+                            throw new AzPSArgumentException(Resources.UpdateKubernetesVersionAndNodeImageOnlyConflict, "KubernetesVersion");
+                        }
+
+                        bool AllAgentPoolVirtualMachineScaleSets = cluster.AgentPoolProfiles.All(c => c.Type.ToLower().Equals("virtualmachinescalesets"));
```
Comment (isra-fel) [inline src/Aks/Aks/Commands/SetAzureRmAks.cs:274]: lower case local variable
```suggestion
                        bool allAgentPoolVirtualMachineScaleSets = cluster.AgentPoolProfiles.All(c => c.Type.ToLower().Equals("virtualmachinescalesets"));
```

**Review by isra-fel — CHANGES_REQUESTED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**